### PR TITLE
chore: add docs for pagination protobuf message

### LINKED ISSUE #8000: Pagination has no doc-strings describing how it is used
### Issue

The proto message for pagination has no doc-strings at all apart from for this one field where it is pointlessly obvious:

```
// Newest records first, older records last, default is true
optional bool newest_first = 5;
```

There should be some doc-strings with some details. 

The pagination struct is used everywhere and so will bloat every API doc if there is too much, therefore it should be kept light; we will have a dedicated docs page talking about how to use pagination.

- [ ] Update the API docs as described above
- [ ] Discuss with @candida getting a new docs page created on how to use pagination (with examples?)

### PR DESCRIPTION
**Pull request by pscott31**
closes #8000

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
...
+// To retrieve subsequent pages the caller needs to examine PageInfo structure returned in the
+// response to find a cursor string which uniquely identifies the last row of that page, and
+// then pass that in the 'after' field of the Pagination message in a subsequent request.
+//
+// To paging backwards, take the cursor for the first row of the page from PageInfo and pass that
+// in the 'before' field of the Pagination message.
+//
+// A pagination message that uses first/after as well as last/before is invalid.
 message Pagination {
+    // The number of records to be returned that sort greater than row identified by cursor supplied in 'after'
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:954]: This line needs to be terminated with a full stop to generate the docs appropriately.

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 2]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
...
     optional int32 first = 1;
+    // If paging forwards, the cursor string for the last row of the previous page.
     optional string after = 2;
+    // The number of records to be returned that sort less than row identified by cursor supplied in 'before'
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:958]: This line needs to be terminated with a full stop to generate docs appropriately.

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
 
 // -- Pagination --
 
-// Cursor information for cursor based pagination requests
+// Pagination information for cursor based pagination.
+//
+// All data returned from the API is ordered in some well defined manner. The details of which
+// columns and in which order the sorting is performed depend on which API is being endpoint is 
+// being called, but usually the timestamp of the block in which it data was last updated is the
+// primary sort column.
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:932]: `depend on which API is being endpoint is being called`: not sure this is making sense.
`the block in which it data`: this too.

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
...
+//
+// In order to prevent exessively large response messages, and to prevent large queries from
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:934]: maybe a typo on exessively?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 5]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
...
+// starving database resources, the API uses a cursor based pagination mechanism.
+//
+// This pagination message is optionally provided as part of the the request to specify
+// - the starting point within a the total result set to begin the page
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:938]: `within a the total`?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 6]
```diff
@@ -924,11 +924,40 @@ service TradingDataService {
...
+// - size of returned page
```
Comment (jeremyletang) [inline protos/sources/data-node/api/v2/trading_data.proto:939]: the size of the returned page?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 7]
```diff
@@ -749,11 +749,39 @@ service TradingDataService {
...
+// To prevent excessively large response messages and to avoid overloading
+// database resources, the API employs a cursor-based pagination mechanism.
+//
+// This Pagination message can be optionally provided as part of the request to specify:
+// - The starting point within the total result set for beginning the page
+// - The size of the returned page
+// - The ordering of the data within that page
+//
+// If no Pagination message is provided, the API will return the first page of
+// data using the default page size. The maximum page size is 1000.
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:766]: Is the maximum page size really 1000? Or was that supposed to say default page size is 1000?

The below gets me 2000 things in `["orders"]["edges"]`:
```
https://api.n00.devnet1.vega.xyz/api/v2/orders?pagination.first=2000
```

Comment (pscott31) [reply protos/sources/data-node/api/v2/trading_data.proto:766]: IIRC we decided that the default should be 100 but the maximum 1000 - though you're right I don't see any code that enforces that!

Comment (pscott31) [reply protos/sources/data-node/api/v2/trading_data.proto:766]: I see here we discussed about the page size, for graphql at least:

https://github.com/vegaprotocol/vega/issues/7291

I'll update the docs to reflect what actually happens at the moment and if/when we decide that we want a limit we can fix the docs up as well

**Review by wwestgarth — APPROVED** [review]

**Review by gordsport — APPROVED** [review]

**Pull request merged**